master/buildbot/steps/shell.py

```diff
@@ -128,9 +127,23 @@ def __init__(self, workdir=None,
 
         # check validity of arguments being passed to RemoteShellCommand
         invalid_args = []
-        signature = inspect.signature(
-            remotecommand.RemoteShellCommand.__init__)
-        valid_rsc_args = signature.parameters.keys()
+        valid_rsc_args = [
```

Comment:
why?

Reply:
We now replace RemoteShellCommand in tests with a wrapper function instead of FakeRemoteShellCommand. inspect can't therefore see the arguments.

ShellMixin also hardcodes the list of arguments, so I think this is actually an improvement.